src/util/ChartUtils.ts

```diff
@@ -976,33 +976,52 @@ export const getStackedData = (
   return stack(data);
 };
 
+type AxisId = string;
+type StackId = string | number | symbol;
+
+export type ParentStackGroup = {
+  hasStack: boolean;
+  stackGroups: Record<StackId, ChildStackGroup>;
+};
+
+export type ChildStackGroup = {
+  numericAxisId: string;
+  cateAxisId: string;
+  items: Array<ReactElement>;
+  stackedData?: ReadonlyArray<Series<Record<string, unknown>, string>>;
+};
+
+export type AxisStackGroups = Record<AxisId, ParentStackGroup>;
+
 export const getStackGroupsByAxisId = (
-  data: any,
-  _items: Array<any>,
+  data: ReadonlyArray<Record<string, unknown>> | undefined,
```

Comment:
this any is so dangerous 😓. What if `data` is `null`?

Eventually we should make `data` generic yeah?

Reply:
The whole library is full of missing null-checks because at the moment, the way typescript is configured it behaves like Java, i.e. each and every type is nullable. We should definitely turn on https://www.typescriptlang.org/tsconfig#strictNullChecks. When I tried, the whole repository lights up with hundreds of errors so I chose to postpone that until later.

Reply:
Gotcha